lib/teleterm/clusters/cluster_kubes.go

```diff
@@ -126,3 +127,37 @@ func (c *Cluster) reissueKubeCert(ctx context.Context, kubeCluster string) error
 		}))
 	}))
 }
+
+func (c *Cluster) getKube(ctx context.Context, kubeCluster string) (types.KubeCluster, error) {
+	var kubeClusters []types.KubeCluster
+	err := AddMetadataToRetryableError(ctx, func() error {
+		proxyClient, err := c.clusterClient.ConnectToProxy(ctx)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		defer proxyClient.Close()
+
+		authClient, err := proxyClient.ConnectToCluster(ctx, c.clusterClient.SiteName)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		defer authClient.Close()
+
+		kubeClusters, err = kubeutils.ListKubeClustersWithFilters(ctx, authClient, proto.ListResourcesRequest{})
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	for _, cluster := range kubeClusters {
+		if cluster.GetName() == kubeCluster {
+			return cluster, nil
+		}
+	}
+	return nil, trace.BadParameter("kubernetes cluster %q not found", kubeCluster)
```

Comment:
It feels like `NotFound` is better suited for that, no?

```suggestion
	return nil, trace.NotFound("kubernetes cluster %q not found", kubeCluster)
```